### PR TITLE
Added tags to S3 objects

### PR DESCRIPTION
**Pull request by codingconcepts**
Hey @Jeffail,

I've added the ability to tag objects in S3. Please give me a shout if you'd like me to make any changes!

Cheers,

Rob

### CODE REVIEW

File: lib/output/writer/s3.go [hunk 1]
```diff
@@ -162,9 +174,17 @@ func (a *AmazonS3) WriteWithContext(wctx context.Context, msg types.Message) err
 			contentEncoding = aws.String(ce)
 		}
 
+		// Prepare tags.
+		tags := []string{}
+		for k, v := range a.tags {
+			tags = append(tags, k+"="+v.String(i, msg))
```
Comment (Jeffail) [inline lib/output/writer/s3.go:180]: I might be wrong as I haven't read up on S3 tags but it looks like we're building a url query string here, so it's probably worth escaping the keys and values with https://golang.org/pkg/net/url/#QueryEscape

File: lib/output/aws_s3.go [hunk 2]
```diff
@@ -123,6 +144,20 @@ calculated per message of a batch.
 
 Metadata fields on messages will be sent as headers, in order to mutate these values (or remove them) check out the [metadata docs](/docs/configuration/metadata).
 
+### Tags
+
+Tag fields will be stored as object tags against the S3 object. For example:
+
+` + "```yaml" + `
+output:
+	aws_s3:
```
Comment (Jeffail) [inline lib/output/aws_s3.go:153]: Looks like your IDE has mixed tabs and spaces, mine does this all the time and it drives me crazy embedding yaml configs in a Go string literal.

File: lib/output/writer/s3.go (hunk 1 above)
Comment (codingconcepts) [reply lib/output/writer/s3.go]: This is fixed, great catch 👍

File: lib/output/aws_s3.go (hunk 2 above)
Comment (codingconcepts) [reply lib/output/aws_s3.go:153]: And lo! My changes seem to have no effect, I'll spend some more time on this tomorrow 😄

Comment (codingconcepts) [reply lib/output/aws_s3.go]: Righto, that's fixed!

**Comment by codingconcepts** [comment]
Apologies, I accidentally resolved the conversation about the query escaping! Please re-open the conversation if you're not happy with the fix, @Jeffail.

**Comment by Jeffail** [comment]
Ace, thanks @codingconcepts, I might make a small tweak as it might be useful to ensure consistent ordering of tags.

**Comment by codingconcepts** [comment]
Awesome, thanks, @Jeffail!

Want me to sort them by keys?

**Comment by Jeffail** [comment]
@codingconcepts, done https://github.com/Jeffail/benthos/commit/228b4377ba13d9b2994efd2aa0c23130706e325c, would be good if you could test it at some point though.

**Pull request merged**